### PR TITLE
Created french keyboardlayout

### PR DESCRIPTION
**Pull request by DaVukovic**
This doesn't exist yet...maybe it's helpfull and a forum user was searching for it.

http://forum.kodi.tv/showthread.php?tid=258151&pid=2231258#pid2231258

### CODE REVIEW

**Comment by stefansaraev** [comment]
looks ok to me.

**Comment by MartijnKaijser** [comment]
http://trac.kodi.tv/ticket/16544

**Comment by 1977David** [comment]
If the other one is better, then ok ;). Didn't saw that on trac.

Use whatever you think it's best.

**Comment by MartijnKaijser** [comment]
I don't mind which is best. Just link the ticket so it could be closed if needed

File: system/keyboardlayouts/french.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+<!--
+Please use English language names instead.
+Default font lacks support for all characters
+-->
+<keyboardlayouts>
+  <layout language="French" layout="AZERTY">
+    <keyboard>
+      <row>1234567890°+</row>
+      <row>azertyuiop^$</row>
+      <row>qsdfghjklmú*</row>
```
Comment (ozolli) [inline system/keyboardlayouts/french.xml]: The ú (acute accent) here is wrong. It should be ù (grave accent).

File: system/keyboardlayouts/french.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+      <row>1234567890</row>
+      <row>AZERTYUIOP</row>
+      <row>QSDFGHJKLM</row>
+      <row>WXCVBN?./</row>
+    </keyboard>
+    <keyboard modifiers="symbol,shift+symbol">
+      <row>)!@#€§$%^&amp;*(</row>
+      <row>[]{}-_=+;:μ</row>
+      <row>'",.&lt;&gt;/?\|</row>
+      <row>`~éèáàúùç</row>
```
Comment (ozolli) [inline system/keyboardlayouts/french.xml]: á and ú are not French symbols.

**Comment by MartijnKaijser** [comment]
@koying can you give some comments as well? thx

**Comment by koying** [comment]
Belgian AZERY keyboards are differents from French ones ;)
But I confirm @ozolli 's remarks

This one is better than the other, although accentuated capitals could be added for good measure

**Comment by MartijnKaijser** [comment]
@ozolli could you provide a fixed layout? Either as PR or pastebin

**Comment by ozolli** [comment]
@MartijnKaijser Here is the pastebin link:
http://pastebin.com/qv2iiHsh

I tried to add as much symbols as possible (all the keyboard layout and almost all lowercase and uppercase French accented characters.

**Comment by 1977David** [comment]
@MartijnKaijser 
I did those changes to the file...

https://github.com/1977David/xbmc/blob/master/system/keyboardlayouts/french.xml

shall I do a new PR?

**Comment by ozolli** [comment]
@1977David 
Please use http://pastebin.com/SeTyMQuv
The one you used is faulty.

**Comment by 1977David** [comment]
@ozolli 
strange...I copied and pasted your pastebin from above to my file at my fork. I did it online and not locally on my machine and push after...no clue why some things after the modiefiers"symbol,shift+smybol" have changed after saving from "&quot;&apos;" to " "' "

And tbh, I can't see any other difference. 

But I'll do another c+p.

**Comment by ozolli** [comment]
@1977David the issue is on line 22 where there is a "<" in the middle of the line preventing it to be loaded by Kodi.
It was replaced by Æ in the last edition.

**Comment by MartijnKaijser** [comment]
Is the one linked good to go?

**Comment by ozolli** [comment]
This one from @1977David is good to go:
https://github.com/1977David/xbmc/blob/master/system/keyboardlayouts/french.xml

**Comment by MartijnKaijser** [comment]
fixed with https://github.com/xbmc/xbmc/commit/2ed49f8eae024eac3394fe20499d60d53b068757
thx

**Pull request merged**